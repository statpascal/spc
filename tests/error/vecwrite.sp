program dummy;

var
    i: integer;
    a: vector of integer;
    b: vector of char;
    c: vector of string;
    d: vector of boolean;
    e: vector of real;

begin
    a := intvec (1, 10);
    b := 'A';
    for i := ord ('B') to ord ('Z') do
	b := combine (b, chr (i));
    c := combine ('vector', 'of', 'string', 'values');
    d := a <= 5;
    e := realvec (0.0, 1.0, 0.1);

    writeln (a);
    writeln (b);
    writeln (c);
    writeln (d);
    writeln (e);

    writeln ('With formatting:');
    writeln (a:5);
    writeln (b:0);
    writeln (c:20);
    writeln (d:10);
    writeln (e:5:2)
end.
