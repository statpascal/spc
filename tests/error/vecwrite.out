1 2 3 4 5 6 7 8 9 10 
A B C D E F G H I J K L M N O P Q R S T U V W X Y Z 
vector of string values 
TRUE TRUE TRUE TRUE TRUE FALSE FALSE FALSE FALSE FALSE 
0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 
With formatting:
    1    2    3    4    5    6    7    8    9   10
ABCDEFGHIJKLMNOPQRSTUVWXYZ
              vector                  of              string              values
      TRUE      TRUE      TRUE      TRUE      TRUE     FALSE     FALSE     FALSE     FALSE     FALSE
 0.00 0.10 0.20 0.30 0.40 0.50 0.60 0.70 0.80 0.90 1.00
